src/languages/scss.js

```diff
@@ -61,13 +66,18 @@ export default function(hljs) {
       },
       {
         className: 'selector-pseudo',
-        begin: ':(visited|valid|root|right|required|read-write|read-only|out-range|optional|only-of-type|only-child|nth-of-type|nth-last-of-type|nth-last-child|nth-child|not|link|left|last-of-type|last-child|lang|invalid|indeterminate|in-range|hover|focus|first-of-type|first-line|first-letter|first-child|first|enabled|empty|disabled|default|checked|before|after|active)'
+        begin: ':(' + PSEUDO_CLASSES.join('|') + ')'
       },
       {
         className: 'selector-pseudo',
-        begin: '::(after|before|choices|first-letter|first-line|repeat-index|repeat-item|selection|value)'
+        begin: '::(' + PSEUDO_ELEMENTS.join('|') + ')'
       },
```

Comment:
While not a CSS best practice, does Sass actually prevent using `:` for pseudo-elements? Looks to me like only the Sass grammar differentiates these.

```js
      {
        className: 'selector-pseudo',
        begin: ':(:)?(' + css_shared.PSEUDO_SELECTORS.join('|') + ')'
      },
```

Or would this not be in the scope of this PR?

Reply:
They are different things according to CSS, so since SCSS broke them out I went ahead and did the same and allowed it both ways.  The grammars that mix them are being slightly LAZY (my first thought).  Sidenote: keep in mind SASS is not SCSS.

_That said I don't necessarily mind sightly lazy._  Much of our parsing is based on what is expedient, not what is 100% correct.  Trying to parse these languages 100% correctly when parsing then 80% correctly will highlight them pretty  much exactly the same is often a battle with no real upsides (only resulting in increased complexity and bugs and maintenance time).

This first pass here is largely about consistency though not perfection in the code.  **First we need good tests, then we can refactor freely.**